tests/golden/th.expected

```diff
@@ -34,7 +34,7 @@ Error in $: not enough input. Expecting json list value
 SomeType (reject unknown fields)
 Error in $: Unknown fields: ["testZero"]
 Error in $: key "tag" not found
-Error in $: Unknown fields: ["testtwo","testone","testthree"]
+Error in $.testone: parsing Double failed, unexpected Boolean
```

Comment:
Same as https://github.com/bos/aeson/pull/774/commits/70d0927aa32481d5f4d312c8c550763fe7d41e27#r419038724 but for TH.

I noticed there where no tests to update to validate the TH bugfix so I created the first commit and extracted it so the reviewer can also see the TH behavior change being reflected in tests.

Reply:
For this comment to make sense, look at the 2nd commit in isolation.